### ISSUE TITLE
swin的预训练权重

### ISSUE THREAD

**Issue by LiBingyu01** [JiaDingCN/GeminiFusion#7]
请问您，swin的预训练权重链接是什么？我在官网上下载的不能匹配进去。、

**Comment by peter-wang321** [comment]
https://github.com/microsoft/Swin-Transformer#main-results-on-imagenet-with-pretrained-models

I used this, it works.

**Comment by peter-wang321** [comment]
Sorry, I also encountered issues like:


unexpected key in source state_dict: norm.module.weight, module.norm.weight, norm.weight, norm.module.bias, module.norm.bias, norm.bias, head.module.weight, module.head.weight, head.weight, head.module.bias, module.head.bias, head.bias, patch_embed.module.proj.weight, patch_embed.module.proj.bias, patch_embed.module.norm.weight, patch_embed.module.norm.bias, patch_embed.proj.weight, patch_embed.proj.bias, patch_embed.norm.module.weight, patch_embed.norm.weight, patch_embed.norm.module.bias, patch_embed.norm.bias, layers.0.blocks.module.1.attn_mask, layers.0.blocks.0.module.norm1.weight, layers.0.blocks.0.module.norm1.bias, layers.0.blocks.0.module.norm2.weight, layers.0.blocks.0.module.norm2.bias, layers.0.blocks.0.module.attn.relative_position_index, layers.0.blocks.0.module.attn.relative_position_bias_table, layers.0.blocks.0.norm1.weight, layers.0.blocks.0.norm1.bias, layers.0.blocks.0.attn.qkv.module.weight, layers.0.blocks.0.attn.qkv.weight, layers.0.blocks.0.attn.qkv.module.bias, layers.0.blocks.0.attn.qkv.bias, layers.0.blocks.0.attn.proj.module.weight, layers.0.blocks.0.attn.proj.weight,

**Comment by peter-wang321** [comment]
> 请问您，swin的预训练权重链接是什么？我在官网上下载的不能匹配进去。、

Have you reproduced the results based on mit_b3 or mit_b5?

**Comment by JiaDingCN** [comment]
> Sorry, I also encountered issues like:
> 
> unexpected key in source state_dict: norm.module.weight, module.norm.weight, norm.weight, norm.module.bias, module.norm.bias, norm.bias, head.module.weight, module.head.weight, head.weight, head.module.bias, module.head.bias, head.bias, patch_embed.module.proj.weight, patch_embed.module.proj.bias, patch_embed.module.norm.weight, patch_embed.module.norm.bias, patch_embed.proj.weight, patch_embed.proj.bias, patch_embed.norm.module.weight, patch_embed.norm.weight, patch_embed.norm.module.bias, patch_embed.norm.bias, layers.0.blocks.module.1.attn_mask, layers.0.blocks.0.module.norm1.weight, layers.0.blocks.0.module.norm1.bias, layers.0.blocks.0.module.norm2.weight, layers.0.blocks.0.module.norm2.bias, layers.0.blocks.0.module.attn.relative_position_index, layers.0.blocks.0.module.attn.relative_position_bias_table, layers.0.blocks.0.norm1.weight, layers.0.blocks.0.norm1.bias, layers.0.blocks.0.attn.qkv.module.weight, layers.0.blocks.0.attn.qkv.weight, layers.0.blocks.0.attn.qkv.module.bias, layers.0.blocks.0.attn.qkv.bias, layers.0.blocks.0.attn.proj.module.weight, layers.0.blocks.0.attn.proj.weight,

Have you tested the released swin transformer models? Do they have the correct accuracy?

**Comment by JiaDingCN** [comment]
It looks like you have modified the model setting code or used the wrong config. You may check whether the code is changed or the running command is right.

**Comment by peter-wang321** [comment]
I have tested all the pre-trained models on the nyuv2 dataset, the swin-large-384 +FineTune from SUN 300eps did not show the correct accuracy, which is just 56.76 instead of 60.9. However, other models have a similar accuracy as the one reported in the Repo.

The running command I used is:
CUDA_VISIBLE_DEVICES=0,1,2 python -m torch.distributed.launch --nproc_per_node=3  --use_env main.py --backbone swin_large_window12 --dataset nyudv2 -c rerun_54.8_swin_large_window12_finetune_dpr0.15_100+200+100 \
 --dpr 0.15 --num-epoch 100 200 100 --is_pretrain_finetune --resume ./swin-large-384.pth.tar --eval --resume ./pretrained/finetune-swin-large-384.pth.tar

Another quick question, I am not sure what the "54.8" means in the "-c" para.

Thanks.

**Comment by JiaDingCN** [comment]
> I have tested all the pre-trained models on the nyuv2 dataset, the swin-large-384 +FineTune from SUN 300eps did not show the correct accuracy, which is just 56.76 instead of 60.9. However, other models have a similar accuracy as the one reported in the Repo.
> 
> The running command I used is: CUDA_VISIBLE_DEVICES=0,1,2 python -m torch.distributed.launch --nproc_per_node=3 --use_env main.py --backbone swin_large_window12 --dataset nyudv2 -c rerun_54.8_swin_large_window12_finetune_dpr0.15_100+200+100 --dpr 0.15 --num-epoch [100 200 100](tel:100 200 100) --is_pretrain_finetune --resume ./swin-large-384.pth.tar --eval --resume ./pretrained/finetune-swin-large-384.pth.tar
> 
> Another quick question, I am not sure what the "54.8" means in the "-c" para.
> 
> Thanks.

You may delete the first —resume. Also, the typo in -c is fixed.

**Comment by JiaDingCN** [comment]
Closed due to inactivity.

**Comment by iacopo97** [comment]
@peter-wang321 Have you managed to obtain the results?

**Comment by peter-wang321** [comment]
yes except the last model.

**Comment by iacopo97** [comment]
@peter-wang321 Did you find the correct checkpoint to train with swin transformer?
I have also tried `use_checkpoint` flag and it will not work, I don't know why

**Comment by peter-wang321** [comment]
@iacopo97 I used the checkpoint from https://github.com/microsoft/Swin-Transformer#main-results-on-imagenet-with-pretrained-models. Most of the results can be re-implemented except the last one, but I think it does not matter.

Best,

**Comment by iacopo97** [comment]
@peter-wang321 Thank you very much. How do you solve the error:
unexpected key in source state_dict: norm.module.weight, module.norm.weight, norm.weight, norm.module.bias, module.norm.bias, norm.bias, head.module.weight, module.head.weight, head.weight, head.module.bias, module.head.bias, head.bias, patch_embed.module.proj.weight, patch_embed.module.proj.bias, patch_embed.module.norm.weight, patch_embed.module.norm.bias, patch_embed.proj.weight, patch_embed.proj.bias, patch_embed.norm.module.weight, patch_embed.norm.weight, patch_embed.norm.module.bias, patch_embed.norm.bias, layers.0.blocks.module.1.attn_mask, layers.0.blocks.0.module.norm1.weight, layers.0.blocks.0.module.norm1.bias, layers.0.blocks.0.module.norm2.weight, layers.0.blocks.0.module.norm2.bias, layers.0.blocks.0.module.attn.relative_position_index, layers.0.blocks.0.module.attn.relative_position_bias_table, layers.0.blocks.0.norm1.weight, layers.0.blocks.0.norm1.bias, layers.0.blocks.0.attn.qkv.module.weight, layers.0.blocks.0.attn.qkv.weight, layers.0.blocks.0.attn.qkv.module.bias, layers.0.blocks.0.attn.qkv.bias, layers.0.blocks.0.attn.proj.module.weight, layers.0.blocks.0.attn.proj.weight,

**Comment by peter-wang321** [comment]
@iacopo97 Sorry, I am not sure if I encountered the same issue, you can find all the issues I mentioned in previous issues. If it is in the list of issues, the solution has been released. If not, I think I did not have the same issue. 
Maybe you can try to use the 
model.load_state_dict(torch.load('model.pth'), strict=False)
to observe the performance first, and then consider the key errors.

That is all what I can provide. Best wishes! More detailed issues, please ask the first author for more help.

**Comment by iacopo97** [comment]
Thank you very much for your help. and your time. I copy the issues that you have at the beginning of the current issue.

**Comment by peter-wang321** [comment]
@iacopo97 Yes, I did. Sorry for missing it. I tried my best to recall the process and the issues I encountered during the re-implementation, but I forgot if I solved this issue. I managed to reproduce the results except the last one. I remembered the most important modification for the code. It is as follows:
# self.cross_attn_0_to_1 = nn.MultiheadAttention(
        #     dim, self.cross_heads, dropout=0.0, batch_first=False
        # )

        self.cross_attn_0_to_1 = nn.MultiheadAttention(
            dim, self.cross_heads, dropout=0.0                  
        )
In this section, I delete the batch_first=False, other changes may be small or still the same.

I should not change other codes, if it is for the last one, I do not know how to do that. But for at least the first two models, it works well.

**Comment by peter-wang321** [comment]
@iacopo97 This is in the path of GeminiFusion/models/swin_transformer.py. I hope you enjoy this model.

**Comment by iacopo97** [comment]
Thank you very much!!!
